snakefile

```diff
@@ -0,0 +1,362 @@
+            --output-fasta {output.fasta} \
+            --output-unix-newline \
+            --output-additional-info {output.additional_info} > {output.flagged_annotation}
+        """
+
+rule transform_genbank:
+    input:
+        "data/genbank.ndjson"
+    output:
+        metadata="data/genbank/metadata.noClade.tsv",
+        fasta="data/genbank/sequences.fasta",
+        problem="data/genbank/problem_data.tsv",
+        flagged_annotation="data/genbank/transform-log.txt",
+        additional_info="data/genbank/additional_info.tsv",
+    shell :
+        '''
+        ./bin/transform-genbank {input} \
+            --output-metadata {output.metadata} \
+            --output-fasta {output.sequences} \
+            --problem-data {output.problem} > {output.flagged_annotation}
+
+        touch {output.additional_info}
+        '''
+
+
+
+
+rule dowload_old_clades :
+    output:
+        "data/{database}/nextclade.old.tsv"
+    params:
+        dst_source=S3_DST+'/nextclade.tsv.gz',
+        src_source='$S3_SRC/nextclade.tsv.gz',
+    shell:
+        '( aws s3 cp --no-progress "{params.dst_source}" - || aws s3 cp --no-progress "{params.src_source}" -) | gunzip -cfq > {output} '
+
+
+rule filter_fasta :
+    input:
+        fasta = lambda wildcards : "data/"+wildcards.database+"/sequences.fasta",
+        tsv = rules.dowload_old_clades.output
+    output:
+        "data/{database}/nextclade.sequences.fasta"
+    shell: 
+        "./bin/filter-fasta --input_fasta=input.fasta --input_tsv=input.tsv --output_fasta={output}" 
+
+rule run_nextclade :
+    input:
+        rules.filter_fasta.output
+    output:
+        "data/{database}/nextclade.new.tsv"
+    shell:
+        "./bin/run-nextclade {input} {output}"
+
+rule join_clades :
+    input:
+        old=rules.dowload_old_clades.output ,
+        new=rules.run_nextclade.output
+    output:
+        "data/{database}/nextclade.tsv"
+    shell:
+        "./bin/join-rows {input.new} {input.old} -o {output}"
+
+
+rule join_metadata_and_clades :
+    input:
+        clades = rules.join_clades.output ,
+        meta = lambda wildcards : "data/"+wildcards.database+"/metadata.noClade.tsv"
+    output:
+        "data/{database}/metadata.tsv"
+    shell:
+        "./bin/join-metadata-and-clades {input.meta} {input.clades} -o {output}"
+
+rule flag_metadata :
+    input :
+        rules.join_metadata_and_clades.output
+    output :
+        "data/{database}/flagged_metadata.txt"
+    shell:
+        "./bin/flag-metadata {input} > {output}"
+
+rule check_locations :
+    input :
+        rules.join_metadata_and_clades.output
+    params:
+        idcolumn=lambda wildcards : config['idcolumn'][wildcards.database]
+    output :
+        "data/{database}/location_hierarchy.tsv"
+    shell:
+        "./bin/check-locations {input} {output} {params.idcolumn}"
+
+
+rule notify_and_upload:
+    input :
+        sequences = lambda wildcards : "data/"+wildcards.database+"/sequences.fasta",
```

Comment:
we can use wildcards directly here
`sequences = "data/{database}/sequences.fasta",`